scripts/irods/test/test_iqstat.py

```diff
@@ -0,0 +1,65 @@
+from __future__ import print_function
+import datetime
+import os
+import re
+import sys
+import json
+
+if sys.version_info < (2, 7):
+    import unittest2 as unittest
+else:
+    import unittest
+
+from . import session
+from ..configuration import IrodsConfig
+from ..test.command import assert_command
+
+class Test_Iqstat(session.make_sessions_mixin([('otherrods', 'rods')], [('alice', 'apass')]), unittest.TestCase):
+
+    def setUp(self):
+        super(Test_Iqstat, self).setUp()
+        self.rule_ids = ''
+        self.admin = self.admin_sessions[0]
+        self.user = self.user_sessions[0]
+
+    def tearDown(self):
+        for rule_id in self.rule_ids.split():
+            self.admin.assert_icommand('iqdel {}'.format(rule_id),'EMPTY')
+        super(Test_Iqstat, self).tearDown()
+
+    def spawn_delay_rule(self,sess):
+        dtnow = datetime.datetime.now()
+        epoch_with_micros = '({0:%s})({0:%f})'.format(dtnow)
+        sess.assert_icommand('''irule -r irods_rule_engine_plugin-irods_rule_language-instance '''
+                             '''"delay('<EF>1m</EF><PLUSET>1s</PLUSET>'){{'{}'}}" null null'''
+                             ''.format(epoch_with_micros), 'EMPTY', desired_rc = 0)
+        rc, rule_id, err = sess.assert_icommand('''iquest "%s" "select RULE_EXEC_ID where RULE_EXEC_NAME like '%{}%'"'''
+                                                    ''.format(epoch_with_micros),'STDOUT',desired_rc = 0)
+        self.rule_ids += rule_id
```

Comment:
You're concatenating strings here. Is a space separating them?

You call `spawn_delay_rule` twice in one test, so I'd expect this line to result in something like `rule_id_1rule_id_2`. The tear down code seems to expect the ids to be delimited in some way.

Am I misunderstanding?

Reply:
`self.rule_ids` actually ends up as `rule_id1\nrule_id2\n`; then in the teardown, `self.rule_ids.split( )`  produces ` [ 'rule_id1', 'rule_id2']`

Reply:
So `+=` means `s = s + '\n' + t` where `s` and `t` are strings?

Reply:
Each rule id returned from the command output includes its own terminating "\n".

Reply:
Maybe better if `rule_id `is rename` rule_id_with_carriage_return` ?